src/devtools/client/debugger/src/utils/editor/hit-counts.ts

```diff
@@ -0,0 +1,47 @@
+import { interpolateLab } from "d3-interpolate";
+import { getLuminance, darken } from "polished";
+import memoize from "lodash/memoize";
+import { HitCount } from "@replayio/protocol";
+
+const hitBadge = ["#FFB5DD", "#DB0069"];
+const text = ["#FFB5DD", "#DB0069"];
+
+export const getHitCountColors = memoize((hitCounts?: HitCount[]) => {
+  if (!hitCounts) {
+    return new Map();
+  }
+
+  const uniqueHitCounts = Array.from(
+    new Set(hitCounts ? hitCounts.map(hitCount => hitCount.hits) : [])
+  );
+  const maxHitCount = Math.max(...uniqueHitCounts);
+
+  const map = new Map(
+    uniqueHitCounts.map(hitCount => {
+      const avg = Math.min(1, Math.max(0, Math.log(hitCount) / Math.log(maxHitCount))) || 0;
+      const backgroundColor = interpolateLab(hitBadge[0], hitBadge[1])(avg);
+      const foregroundColor =
+        getLuminance(backgroundColor) > 0.5 ? "rgba(0,0,0,0.97)" : "rgba(255,255,255,.99)";
```

Comment:
I don't understand the very specific but also kind of unexpected values of `rgba(0,0,0,0.97)` and `rgba(255,255,255,.99)`. Why .97 and .99?

Reply:
yeah - i'm not sure I follow either. this came from elsewhere

Reply:
I think we might as well go with `#fff` and `#000` then 😁